main.cpp

```diff
@@ -121,368 +155,403 @@ struct CarWash
+    }
+    else
+    {
+        sampleStart = 0;
+        sampleEnd = 1;
+    }
+}
+
+void SolideStateAmp::imitateTubeAmp(int ampSelectIndex, std::string speakerIr)
+{
+    if (ampSelectIndex <= 50 && speakerIr == "Celestion")
+    {
+        numSpeakers = 4;
+        speakerSize = 12.0f;
+    }
+    else
+    {
+        numSpeakers = 2;
+        speakerSize = 10.0f; 
+    }
+}
+
+void SolideStateAmp::amplifieSound(int inputSig, float gain)
+{
+    masterLvl = (inputSig * gain)/100;
+}
 
 struct SamplingPad
 {
-    //number of pads
     int numPads = 8;
-    //number of intergrated instruments
     int numIntergratedInst = 25; 
-    //stereo or mono output
     bool stereoOut = true; 
-    //amount of sampling  memory space in sec
     int samplingSpaceSec = 120;
-    //number of integrated effects
     int numIntegratedFx = 12;
 
-    //trigger  .wav sounds
-    void trigSnd (int padIndex, bool trig);
-    //play a sequence
+    void trigSnd ();
     void playSequence (int numSteps, int bpm, bool startSequence);
-    //send midi
     void sendMidi (int midiNoteNum, int midiVel, bool noteEvent, bool isCc);
 };
 
-/*
-Thing 4) analog console 
-5 properties:
-    1) number of channel strips
-    2) number of audio send
-    3) number of monitor out
-    4) number of steroe chanels
-    5) pre or post EQ selector
-3 things it can do:
-    1) Mix different sounds together 
-    2) contrôl master output level
-    3) add tone
- */
+void SamplingPad::trigSnd(){}
+
+void SamplingPad::playSequence(int numSteps, int bpm, bool startSequence)
+{
+    if (startSequence == 1)
+    {
+        numSteps *= bpm;
+    }
+    else
+    {
+        numSteps = 0;
+    }
+}
+
+void SamplingPad::sendMidi(int midiNoteNum, int midiVel, bool noteEvent, bool isCc)
+{
+    midiNoteNum = 127;
+    midiVel = 127;
+    noteEvent = true;
+    isCc = false;
+}
 
 struct AnalogConsole
 {
-    //number of channel strips
     int numChannelStrips = 24;
-    //number of audio send
     int numAudioSend = 2; 
-    //number of monitor out
     int numMonitorOut = 4;
-    //number of steroe chanels
     int numStereoChanels = 2;
-    //pre or post EQ selector
     bool preFader = false;
 
-    //Mix different sounds together
     void mixSounds (int trackNum, float trackLvl, float pan); 
-    //contrôl master output level
     void masterOut (float masterOutdB);
-    //add tone
     void addTone (float hiFreq, float hiBoost, float midFreq, float midBoost, float bassFreq, float bassBoost);
 };
 
-/*
-Thing 5) Mixer
-5 properties:
-    1) maximum number of faders 
-    2) Pan knob middle position value 
-    3) record btn diameter 
-    4) slider length
-    5) number of inserts 
-3 things it can do:
-    1) record audio on track
-    2) pan sound
-    3) change audio volume  
- */
+void AnalogConsole::mixSounds(int trackNum, float trackLvl,float pan)
+{
+    trackNum = 1;
+    trackLvl = -6.0f;
+    pan = 0;
+}
+
+void AnalogConsole::masterOut(float masterOutdB = -6.0f)
```

Comment:
default parameters are normally supplied in the header, not the implementation.
this is because early libraries would distribute a compiled file and a header file, so people trying to use the library would need to look at the header file to figure out how to use it.
- [ ] revise